cmd/permissions-checker/main.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type defaultPermissions struct {
+	file  string
+	dir   string
+	owner string
+	group string
+}
+
+type searchSettings struct {
+	hidden        bool
+	exclude       string
+	baseDirectory string
+}
+
+var (
+	// VERSION will be overwritten automatically by the build system
+	VERSION = "devel"
+)
+
+func main() {
+	var (
+		getVersion  bool
+		defaultPerm defaultPermissions
+		search      searchSettings
+		verbose     bool
+	)
+
+	flag.StringVar(&defaultPerm.file, "file_default", "rw-rw-r--", "Default file permissions")
+	flag.StringVar(&defaultPerm.dir, "dir_default", "rwxrwxr-x", "Default directory permissions")
+	flag.StringVar(&defaultPerm.owner, "owner", "bitnami", "Default owner")
+	flag.StringVar(&defaultPerm.group, "group", "daemon", "Default group")
+	flag.StringVar(&search.baseDirectory, "dir", "/opt/bitnami", "Directory to check")
+	flag.StringVar(&search.exclude, "exclude", "(?!.*)", "Files and/or directories to be excluded")
+	flag.BoolVar(&search.hidden, "hidden", false, "Includes hidden files and directories")
+	flag.BoolVar(&verbose, "verbose", false, "Print every file and directory")
+	flag.BoolVar(&getVersion, "version", false, "Show current version")
+	flag.Parse()
+
+	// Unifies the format eliminating the last "/" if exists
+	search.baseDirectory = strings.TrimSuffix(search.baseDirectory, "/")
+	search.exclude = strings.TrimSuffix(search.exclude, "/")
+
+	// Checks if the default permissions introduced by the user are in the Linux format
+	if m, _ := regexp.MatchString("^[-rwx]{9,10}$", defaultPerm.file); !m {
+		log.Fatalf("file_default should be in the Linux format (i.e. \"rw-rw-r--\")\n")
```

Comment:
no need for the trailing newline in the formats passed to the `log` methods.

consider using simply```
log.Fatalf(`file_default should be in the Linux format (i.e. "rw-rw-r--")`)
```